### ISSUE TITLE
etcd v3 support 

### ISSUE THREAD

**Issue by Jarven89** [bketelsen/crypt#6]
our team is using etcd3 now, can add etcd v3 support?

**Comment by sagikazarmark** [comment]
I think this is becoming more of an issue. `crypt` is currently uninstallable when importing the etcd backend:

```go
package main

import _ "github.com/bketelsen/crypt/backend/etcd"

func main() {

}
```

It dies with:

```
go: finding module for package go.etcd.io/etcd/pkg/fileutil
go: finding module for package github.com/coreos/pkg/capnslog
go: finding module for package github.com/soheilhy/cmux
go: finding module for package github.com/grpc-ecosystem/go-grpc-middleware
go: finding module for package gopkg.in/yaml.v2
go: finding module for package github.com/prometheus/client_golang/prometheus/promhttp
go: finding module for package github.com/dustin/go-humanize
go: finding module for package github.com/jonboulle/clockwork
go: finding module for package github.com/google/uuid
go: finding module for package github.com/grpc-ecosystem/go-grpc-prometheus
go: finding module for package github.com/gogo/protobuf/proto
go: finding module for package github.com/prometheus/client_golang/prometheus
go: finding module for package github.com/golang/groupcache/lru
go: finding module for package github.com/grpc-ecosystem/grpc-gateway/runtime
go: finding module for package github.com/gogo/protobuf/gogoproto
go: finding module for package sigs.k8s.io/yaml
go: finding module for package github.com/coreos/bbolt
go: finding module for package github.com/coreos/go-systemd/journal
go: finding module for package github.com/tmc/grpc-websocket-proxy/wsproxy
go: finding module for package go.uber.org/zap/zapcore
go: finding module for package github.com/grpc-ecosystem/grpc-gateway/utilities
go: finding module for package github.com/prometheus/client_model/go
go: finding module for package github.com/dgrijalva/jwt-go
go: finding module for package github.com/xiang90/probing
go: finding module for package go.uber.org/zap
go: found github.com/coreos/pkg/capnslog in github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
go: found github.com/soheilhy/cmux in github.com/soheilhy/cmux v0.1.4
go: found go.etcd.io/etcd/pkg/fileutil in go.etcd.io/etcd v3.3.24+incompatible
go: found gopkg.in/yaml.v2 in gopkg.in/yaml.v2 v2.3.0
go: found github.com/google/uuid in github.com/google/uuid v1.1.1
go: found go.uber.org/zap in go.uber.org/zap v1.15.0
go: found github.com/grpc-ecosystem/go-grpc-prometheus in github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
go: found github.com/grpc-ecosystem/grpc-gateway/runtime in github.com/grpc-ecosystem/grpc-gateway v1.14.7
go: found github.com/tmc/grpc-websocket-proxy/wsproxy in github.com/tmc/grpc-websocket-proxy v0.0.0-20200427203606-3cfed13b9966
go: found sigs.k8s.io/yaml in sigs.k8s.io/yaml v1.2.0
go: found github.com/gogo/protobuf/proto in github.com/gogo/protobuf v1.3.1
go: found github.com/prometheus/client_golang/prometheus in github.com/prometheus/client_golang v1.7.1
go: found github.com/jonboulle/clockwork in github.com/jonboulle/clockwork v0.2.0
go: found github.com/dustin/go-humanize in github.com/dustin/go-humanize v1.0.0
go: found github.com/grpc-ecosystem/go-grpc-middleware in github.com/grpc-ecosystem/go-grpc-middleware v1.2.1
go: found github.com/xiang90/probing in github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2
go: found github.com/coreos/go-systemd/journal in github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf
go: found github.com/dgrijalva/jwt-go in github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: found github.com/coreos/bbolt in github.com/coreos/bbolt v1.3.5
go: found github.com/golang/groupcache/lru in github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: found github.com/prometheus/client_model/go in github.com/prometheus/client_model v0.2.0
go: dsa imports
	github.com/bketelsen/crypt/backend/etcd imports
	github.com/coreos/etcd/client tested by
	github.com/coreos/etcd/client.test imports
	github.com/coreos/etcd/integration imports
	github.com/coreos/etcd/etcdserver imports
	github.com/coreos/etcd/mvcc/backend imports
	github.com/coreos/bbolt: github.com/coreos/bbolt@v1.3.5: parsing go.mod:
	module declares its path as: go.etcd.io/bbolt
	        but was required as: github.com/coreos/bbolt
```

The combination of vendoring and package relocation doesn't work really well.

**Comment by bketelsen** [comment]
closed by #11
